data/data-pipeline/data_pipeline/etl/sources/epa_rsei_aggregate/etl.py

```diff
@@ -0,0 +1,268 @@
+            field_names.EPA_RSEI_SCORE_OUTPUT_FIELD,
+            field_names.EPA_RSEI_CSCORE_OUTPUT_FIELD,
+            field_names.EPA_RSEI_NCSCORE_OUTPUT_FIELD,
+            field_names.EPA_RSEI_POPULATION_OUTPUT_FIELD,
+            field_names.EPA_RSEI_SCORE_PERCENTILE_LABEL_FIELD,
+            field_names.EPA_RSEI_NCSCORE_PERCENTILE_LABEL_FIELD,
+            field_names.EPA_RSEI_CSCORE_PERCENTILE_LABEL_FIELD,
+            field_names.EPA_RSEI_SCORE_PERCENTILE_RANK_FIELD,
+            field_names.EPA_RSEI_SCORE_THRESHOLD_FIELD,
+        ]
+
+        self.df: pd.DataFrame
+
+    def _custom_percentile(self, data, percentile):
+        """Custom percentile function for any univariate distribution
+        Args:
+            data (pd.Series): Sample distribution and the percentile that we want to calculate
+            percentile (float): The percentile to compute
+        Returns:
+            output (float): Value for the percentile
+        """
+        if percentile == 100:
+            raise ValueError("Percentile cannot be 100")
+
+        n = len(data)
+        p = n * percentile / 100
+        output = None
+        # if integer search for the p-th values in our distribution (sorted in ascending order)
+        if p.is_integer():
+            output = sorted(data)[int(p)]
+        # if this else clause is not added one will get an error
+        # that says that the index value of the list should be an integer number
+        else:
+            output = sorted(data)[int(math.ceil(p)) - 1]
+        return output
+
+    def extract(self) -> None:
+        logger.info("Starting 2.5 MB data download.")
+
+        unzip_file_from_url(
+            file_url=self.AGGREGATED_RSEI_SCORE_FILE_URL,
+            download_path=self.TMP_PATH,
+            unzipped_file_path=self.TMP_PATH / "epa_rsei_aggregated_microdata",
+        )
+
+        self.df = pd.read_csv(
+            filepath_or_buffer=self.TMP_PATH
+            / "epa_rsei_aggregated_microdata"
+            / "CensusMicroTracts2019_2019_aggregated.csv",
+            # The following need to remain as strings for all of their digits, not get
+            # converted to numbers.
+            low_memory=False,
+            header=0,
+        )
+
+    def transform(self) -> None:
+        logger.info("Starting transforms.")
+
+        # the column headers from the above dataset are actually a census tract's data at this point
+        # We will use this data structure later
+        input_columns = [
+            self.TRACT_INPUT_COLUMN_NAME,
+            self.NUMBER_FACILITIES_INPUT_FIELD,
+            self.NUMBER_RELEASES_INPUT_FIELD,
+            self.NUMBER_CHEMICALS_INPUT_FIELD,
+            self.AVERAGE_TOXICITY_INPUT_FIELD,
+            self.SCORE_INPUT_FIELD,
+            self.POPULATION_INPUT_FIELD,
+            self.CSCORE_INPUT_FIELD,
+            self.NCSCORE_INPUT_FIELD,
+        ]
+        # The subsequent step make the current columns the first row
```

Comment:
Nit *steps

Reply:
should still be "steps"?

Reply:
This is a bit of an odd and complicated way to do what you're trying to achieve here. 

Why not just write, in the `extract` method: 

```
input_columns = [
            self.TRACT_INPUT_COLUMN_NAME,
            self.NUMBER_FACILITIES_INPUT_FIELD,
            self.NUMBER_RELEASES_INPUT_FIELD,
            self.NUMBER_CHEMICALS_INPUT_FIELD,
            self.AVERAGE_TOXICITY_INPUT_FIELD,
            self.SCORE_INPUT_FIELD,
            self.POPULATION_INPUT_FIELD,
            self.CSCORE_INPUT_FIELD,
            self.NCSCORE_INPUT_FIELD,
        ]
self.df = pd.read_csv(
            filepath_or_buffer=self.TMP_PATH
            / "epa_rsei_aggregated_risk_environmental_indicators"
            / "CensusMicroTracts2019_2019_aggregated.csv",
            # The following need to remain as strings for all of their digits, not get
            # converted to numbers.
            low_memory=False,
            names=input_columns,
        )
```

And then you don't need any of this complex combinations?

Reply:
@saran-ahluwalia , why did you mark this as resolved?

Reply:
Woah! I appreciate the fresh perspective @lucasmbrown-usds. I got so hung up on looking at the [data](https://github.com/usds/justice40-tool/issues/1070#issuecomment-1005604014)  that I didn't see the forest for the trees.